frontend/src/resetUsername.js

```diff
@@ -1,37 +1,117 @@
-import {Logo, Textbox, FormButton} from "./resetPswd.js";
+import {useState, useEffect} from 'react';
 import FormCSS from "./form.module.css";
+import axios from 'axios';
+
+
 
 export default function ResetUsernameForm() {
+    const [email, setEmail] = useState('');
+    const [username, setUsername] = useState('');
+    const [newUsername, setNewUsername] = useState('');
+    const [message, setMessage] = useState('');
+
+    const checkUsername = async () => {
+        try {
+            const response = await axios.get(`http://localhost:8080/users/${newUsername}`)
+            return response.data.exists;
+        } catch (err) {
+            console.error('error checking username: ', err)
+            return false;
+        }
+    }
+
+    const handleSubmit = async (e) => {
+        e.preventDefault();
+        setMessage('');
+        const usernameExists = await checkUsername();
+        if (!email || !username || !newUsername) {
```

Comment:
This logic should go above the line right above it since we want to make sure the inputs are filled before we even check if the username exists already in the database

Reply:
fixed